engines/osxnative/user_test.go

```diff
@@ -0,0 +1,58 @@
+package osxnative
+
+import (
+	"os"
+	osuser "os/user"
+	"testing"
+)
+
+func TestUser(t *testing.T) {
+	currentUser, err := osuser.Current()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if currentUser.Username != "root" {
+		t.Skip("This test requires root privileges, skipping.")
+	}
+
+	u := user{}
+	err = u.create()
+	if err != nil {
```

Comment:
one thing that's kind of nice is the assert library I've used elsewhere that you could do:
assert.Nil(t, err, "Couldn't create user")